src/test/storage/iterables_test.cpp

```diff
@@ -103,139 +108,197 @@ class IterablesTest : public BaseTest {
-}
+class EncodedSegmentIterablesTest : public IterablesTest,
+                                    public ::testing::WithParamInterface<std::tuple<SegmentEncodingSpec, bool, bool>> {
+ public:
+  uint32_t expected_sum(bool nullable, bool with_position_filter) {
+    if (nullable && with_position_filter) {
+      return 12480u;
+    } else if (nullable && !with_position_filter) {
+      return 24825u;
+    } else if (!nullable && with_position_filter) {
+      return 13579u;
+    } else {
+      return 13702u;
+    }
+  }
 
-TEST_F(IterablesTest, ValueSegmentNullableIteratorWithIterators) {
-  const auto chunk = table_with_null->get_chunk(ChunkID{0u});
+  std::vector<ChunkOffset> expected_offsets(bool with_position_filter) {
+    if (with_position_filter) {
+      return std::vector<ChunkOffset>{ChunkOffset{0}, ChunkOffset{1}, ChunkOffset{2}};
+    } else {
+      return std::vector<ChunkOffset>{ChunkOffset{0}, ChunkOffset{1}, ChunkOffset{2}, ChunkOffset{3}};
+    }
+  }
+};
 
-  auto segment = chunk->get_segment(ColumnID{0u});
-  auto int_segment = std::dynamic_pointer_cast<const ValueSegment<int>>(segment);
+auto formatter = [](const ::testing::TestParamInfo<std::tuple<SegmentEncodingSpec, bool, bool>> info) {
+  auto stream = std::stringstream{};
+  stream << std::get<0>(info.param) << (std::get<1>(info.param) ? "WithNulls" : "")
+         << (std::get<2>(info.param) ? "WithFilter" : "");
 
-  auto iterable = ValueSegmentIterable<int>{*int_segment};
+  auto string = stream.str();
+  string.erase(std::remove_if(string.begin(), string.end(), [](char c) { return !std::isalnum(c); }), string.end());
 
-  auto sum = uint32_t{0};
-  auto accessed_offsets = std::vector<ChunkOffset>{};
-  iterable.with_iterators(SumUpWithIterator{sum, accessed_offsets});
+  return string;
+};
 
-  EXPECT_EQ(sum, 13'702u);
-  EXPECT_EQ(accessed_offsets,
-            (std::vector<ChunkOffset>{ChunkOffset{0}, ChunkOffset{1}, ChunkOffset{2}, ChunkOffset{3}}));
+INSTANTIATE_TEST_SUITE_P(SegmentEncoding, EncodedSegmentIterablesTest,
+                         ::testing::Combine(::testing::ValuesIn(all_segment_encoding_specs),
+                                            ::testing::Bool(),   // nullable
+                                            ::testing::Bool()),  // position filter
+                         formatter);
+
+TEST_P(EncodedSegmentIterablesTest, IteratorWithIterators) {
+  auto encoding_spec = std::get<0>(GetParam());
+  auto nullable = std::get<1>(GetParam());
+  auto with_position_filter = std::get<2>(GetParam());
+  std::shared_ptr<Table> test_table = (nullable ? table : table_with_null);
+
+  auto chunk_encoding_spec = ChunkEncodingSpec{test_table->column_count(), EncodingType::Unencoded};
+  for (auto column_id = ColumnID{0}; column_id < test_table->column_count(); ++column_id) {
+    if (encoding_supports_data_type(encoding_spec.encoding_type, test_table->column_data_type(column_id))) {
```

Comment:
Could we skip if the encoding is not supported? We'll run a couple of duplicated tests with Unencoded here, right?

Reply:
i found that `GTEST_SKIP()` can do this. however this produces "skipped tests" in the test resume. is this what we want? the alternative would be to specify the parameter combinations manually and exclude string-specific encodings.

Reply:
@mrks & @Bensk1  : your take unnecessary tests and how to skip them best?

Reply:
> I don't like the use of GTEST_SKIP() here either. I can live with a simple return;

Agreed